strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseController.java

```diff
@@ -0,0 +1,163 @@
+                                               @PathVariable("storageId") String storageId)
+    {
+        try
+        {
+            Storage storage = configurationManager.getConfiguration().getStorage(storageId);
+            if (storage == null) 
+            {
+                return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                        .body("{ 'error': 'The requested storage was not found.' }");
+            }
+            
+            Map<String,SortedSet<String>> repos = new HashMap<>();
+            repos.put("repositories", new TreeSet<String>(storage.getRepositories().keySet()));
+            
+            return ResponseEntity.ok(new ObjectMapper().writeValueAsString(repos));
+        }
+        catch (Exception e)
+        {
+            logger.error(e.getMessage(), e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(String.format("{ 'error': '%s' }", e.getMessage()));
+        }
+    }
+    
+    @ApiOperation(value = "List the contents for a repository.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "The list was returned."),
+                            @ApiResponse(code = 404, message = "The requested storage, repository, or path was not found."),
+                            @ApiResponse(code = 400, message = "An error occurred.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(value= {"/{storageId}/{repositoryId}","/{storageId}/{repositoryId}/{path}"}, method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> contents(@ApiParam(value = "The storageId", required = true)
+                                           @PathVariable("storageId") String storageId,
+                                           @ApiParam(value = "The repositoryId", required = true)
+                                           @PathVariable("repositoryId") String repositoryId,
+                                           @ApiParam(value = "The repository path", required = false)
+                                           @PathVariable("path") Optional<String> path)
+    {
+        try
+        {
+            Storage storage = configurationManager.getConfiguration().getStorage(storageId);
+            if (storage == null) 
+            {
+                return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                        .body("{ 'error': 'The requested storage was not found.' }");
+            }
+            
+            Repository repository = storage.getRepository(repositoryId);
+            if (repository == null) 
+            {
+                return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                        .body("{ 'error': 'The requested repository was not found.' }");
+            }
+            
+            File baseDir = new File(repository.getBasedir()).getAbsoluteFile();
```

Comment:
`File baseDir = new File(repository.getBasedir()).getAbsoluteFile();`

We like `Path` from `java.nio.file` We don't like `java.io.File`

Reply:
We do as well. The line was only doing path manipulation, but I'll fix it anyway